File: android/src/main/java/com/wxwx/flutter_alibc/PluginResponse.java
package com.wxwx.flutter_alibc;

import java.io.Serializable;
import java.util.HashMap;
import java.util.Map;

/**
 * @Author karedem
 * @Date 2019/9/7 19:55
 * @Description 插件 响应返回实体类
**/
public class PluginResponse implements Serializable {

    private String errorCode;
    private String errorMessage;
    private Object data;

    public static PluginResponse success(Object obj){
        return new PluginResponse("0", "成功", obj);
    }

    public PluginResponse(String errorCode, String errorMessage, Object data) {
        this.errorCode = errorCode;
        this.errorMessage = errorMessage;
        this.data = data;
    }

    public Map<String, Object> toMap(){
        HashMap<String, Object> map = new HashMap<>();
        map.put("errorCode", errorCode);
        map.put("errorMessage", errorMessage);
        map.put("data", data);
        return map;
    }
}

File: android/src/main/kotlin/com/wxwx/flutter_alibc/PluginResponse.kt
package com.wxwx.flutter_alibc

import java.util.*

class PluginResponse constructor(var errorCode: String?, var errorMessage: String?, var data: Any?){

    companion object{
        fun success(obj: Any?): PluginResponse {
            return PluginResponse("0", "成功", obj)
        }
    }

    fun toMap(): Map<String, Any?>? {
        val map = HashMap<String, Any?>()
        map["errorCode"] = errorCode
        map["errorMessage"] = errorMessage
        map["data"] = data
        return map
    }
}